tutorials/Optimize.qmd

```diff
@@ -156,9 +155,62 @@ Then we reach approximately the same point as in the previous run, but in far le
 [f(M, m3)-f(M,m4), distance(M, m3, m4)]
 ```
 
+## Using the “Tutorial” mode
+
+Since a few things on manifolds are a bit different from (classical) Euclidean optimization, `Manopt.jl` has a mode to warn about a few pitfalls.
+
+It can be set using
+
+```{julia}
+#| warning: true
+set_manopt_parameter!(:Mode, "Tutorial")
+```
+
+to activate these. Continuing from the example before, one might argue, that the
+minimizer of $f$ does not depend on the scaling of the function.
+In theory this is of course also the case on manifolds, but for the optimizations there is a caveat. When we define the Riemannian mean without the scaling
+
+```{julia}
+f2(M, p) = sum(1 / 2 * distance.(Ref(M), Ref(p), data) .^ 2)
+grad_f2(M, p) = sum(grad_distance.(Ref(M), data, Ref(p)));
+```
+
+And we consider the gradient at the starting point in norm
+
+```{julia}
+norm(M, data[1], grad_f2(M, data[1]))
+```
+
+On the sphere, when we follow a geodesic, we “return” to the start point after length $2π$. If we “land” short before the starting point due to a gradient of length just shy of $2π$, line search would not make much sense probably.
```

Comment:
Line search will always return something but it won't take into account that gradient is just a local approximation of the objective.

I think it would be better to rephrase it so that you don't have to use the word "probably".

Reply:
You are right, it will always return something, but in the situation described, the step size might actually just be 0, since for such a long vector -grad is basically a very scaled down +grad .

I will try to rephrase this to avoid the probably though.